Sming/SmingCore/Platform/System.cpp

```diff
@@ -11,15 +11,26 @@
 SystemClass System;
 
 os_event_t SystemClass::taskQueue[TASK_QUEUE_LENGTH];
+volatile uint8_t SystemClass::taskCount;
+volatile uint8_t SystemClass::maxTaskCount;
 
 /** @brief OS calls this function which invokes user-defined callback
  *  @note callback function pointer is placed in event->sig, with parameter
  *  in event->par.
  */
-static void taskHandler(os_event_t* event)
+void SystemClass::taskHandler(os_event_t* event)
 {
 	auto callback = reinterpret_cast<TaskCallback*>(event->sig);
 	if(callback) {
+		/*
+		 * @todo how to guarantee atomic operation?
```

Comment:
Atomic operations are possible in ESP8266. You can take a look here: 
https://github.com/wizard97/Embedded_RingBuf_CPP/blob/9e091e9632bc3eba513f422ee7cb17895c565a1a/RingBufHelpers.h#L24

 I remember also using [GCCs __cleanup](http://echorand.me/site/notes/articles/c_cleanup/cleanup_attribute_c.html) but have to scratch my head a bit harder figure out which was the project...

Reply:
https://github.com/wizard97/SimplyAtomic/blob/master/esp8266.h

Reply:
Thanks. It's a bit heavy! I was hoping there was an intrinsic to do a simple decrement rather than have to disable interrupts... I'm inclined to just leave it as-is, it's only for debugging and system info so not critical...